src/contracts/proxies/Vault721.sol

```diff
@@ -195,6 +253,19 @@ contract Vault721 is ERC721EnumerableUpgradeable {
     if (from != address(0)) {
       address payable proxy;
 
+      if (_allowlist[msg.sender]) {
+        if (
+          block.number < _hashState[firstTokenId].lastBlockNumber + blockDelay
+            || _hashState[firstTokenId].lastHash != uint256(nftRenderer.getStateHashBySafeId(firstTokenId))
+        ) {
+          revert BlockDelayNotOver();
+        }
+      } else {
+        if (block.timestamp < _hashState[firstTokenId].lastBlockTimestamp + timeDelay) {
+          revert TimeDelayNotOver();
+        }
+      }
```

Comment:
i'm not fully understanding this:
if whitelisted, it reverts with either:
- block delay not passed
- hash different than lastHash

if NOT whitelisted, it only reverts with:
- time delay not passed

why is this difference? what's the point of "whitelisting" an address, to my understanding is:
"this marketplace is whitelisted, so we check the hash on their transfers"
"this marketplace is NOT whitelisted, so we only check timestamp has passed"

Reply:
so the previous hash and current hash are always compared. however, on non-allowlisted markets, a time-delay of n time is enforced before transferFrom can execute, but on allowlisted markets, a block-delay of 1 or more blocks is enforced because hashes are not only compared internally, but also with the market that checks the hash in the JSON description to ensure that the frontend display of the NFV matches the previous and current hash.

does that clarify it?

Reply:
basically, allowlisted markets don't have a noticeable delay since the block times on Arbitrum are about 5 seconds or so, which is good for UX